modules/dnn/CMakeLists.txt

```diff
@@ -2,26 +2,21 @@ if(WINRT)
   ocv_module_disable(dnn)
 endif()
 
-if(DEFINED BUILD_opencv_dnn AND NOT BUILD_opencv_dnn)
-  return()
-endif()
-
-include(${OpenCV_SOURCE_DIR}/cmake/OpenCVFindLibProtobuf.cmake)
-if(NOT Protobuf_FOUND)
+if(NOT HAVE_PROTOBUF)
   ocv_module_disable(opencv_dnn)
 endif()
 
 set(the_description "Deep neural network module. It allows to load models from different frameworks and to make forward pass")
+ocv_add_module(dnn opencv_core opencv_imgproc WRAP python matlab java js)
 
 ocv_add_dispatched_file("layers/layers_common" AVX AVX2 AVX512_SKX)
 
-ocv_add_module(dnn opencv_core opencv_imgproc WRAP python matlab java js)
```

Comment:
Don't reorder `ocv_add_dispatched_file()` and `ocv_add_module()` (and especially `ocv_define_module()`)